pulp_smash/tests/rpm/api_v2/test_orphan_remove.py

```diff
@@ -1,59 +1,176 @@
 from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
 
 
-def _count_orphans(client):
-    """Count the number of orphans."""
-    orphan_dict = client.get(ORPHANS_PATH)
-    total_orphans = 0
-    for _, data in orphan_dict.items():
-        total_orphans += data.get('count') or 0
-    return total_orphans
+def _count_orphans(orphans):
+    """Count the total number of orphans across all content types."""
+    return sum(content_type['count'] for content_type in orphans.values())
+
+
+class OrphansTestCase(utils.BaseAPITestCase):
+    """Establish that API calls related to orphans function correctly.
 
+    At a high level, this test case does the following:
 
-class OrphanRemoveAllTestCase(utils.BaseAPITestCase):
-    """Establish that orphan removal does a full cleanup."""
+    1. Create an RPM repository with a feed and sync in content. Delete the
+       repository, thus leaving behind orphans.
+    2. Make an API call related to orphans, and assert that the call had the
+       desired effect. Repeat as needed.
+
+    .. NOTE:: Though the test_* methods must execute in a specific order, they
+        (should) all function correctly no matter how many other test_* methods
+        fail.
+    """
 
     @classmethod
     def setUpClass(cls):
-        """Create an RPM repo, sync it, delete the repo, remove orphans."""
-        super(OrphanRemoveAllTestCase, cls).setUpClass()
+        """Create, sync and delete an RPM repository.
+
+        Doing this provides orphans that the remaining test methods can make
+        use of. If this method fails, it's possible that other repositories
+        exist with references to the same content units.
+        """
+        super(OrphansTestCase, cls).setUpClass()
+        cls.setupclass_failure = None
+
+        # Create orphans.
         client = api.Client(cls.cfg, api.json_handler)
         body = gen_repo()
         body['importer_config']['feed'] = RPM_FEED_URL
         repo = client.post(REPOSITORY_PATH, body)
-        utils.sync_repo(cls.cfg, repo['_href'])
+        try:
+            utils.sync_repo(cls.cfg, repo['_href'])
+        finally:
+            client.delete(repo['_href'])
 
-        cls.num_orphans_pre_repo_del = _count_orphans(client)
-        client.delete(repo['_href'])
-        cls.num_orphans_post_repo_del = _count_orphans(client)
-        client.delete(ORPHANS_PATH)
-        cls.num_orphans_after_rm = _count_orphans(client)
+        # Verify that orphans are present. Support for langpack content units
+        # was added in Pulp 2.9.
+        orphans = client.get(ORPHANS_PATH)
+        expected_count = 39
+        if cls.cfg.version >= Version('2.9'):
+            expected_count += 1
+        actual_count = _count_orphans(orphans)
+        if expected_count != actual_count:
+            # We can't use fail(), as it's an instance method.
+            cls.setupclass_failure = (
```

Comment:
`raise AssertionError(msg)` doesn't work? self.fail later on will translate to this [1].

[1] https://hg.python.org/unittest2/file/tip/unittest2/case.py#l688